api/src/pcapi/core/offers/api.py

```diff
@@ -1057,3 +1058,47 @@ def cancel_educational_offer_booking(offer: Offer) -> None:
             "Could not notify offerer about deletion of stock",
             extra={"stock": stock.id},
         )
+
+
+def create_educational_shadow_stock_and_set_offer_showcase(
+    stock_data: PostEducationalOfferShadowStockBodyModel, user: User, offer_id: str
+) -> Stock:
+    beginning = datetime.datetime(2030, 1, 1)
+    booking_limit_datetime = datetime.datetime(2030, 1, 1)
+    total_price = 1
+    number_of_tickets = 1
+    educational_price_detail = stock_data.educational_price_detail
+
+    offer = Offer.query.filter_by(id=offer_id).options(joinedload(Offer.stocks)).one()
+    if len(offer.activeStocks) > 0:
+        raise educational_exceptions.EducationalStockAlreadyExists()
+
+    if not offer.isEducational:
+        raise educational_exceptions.OfferIsNotEducational(offer_id)
+    validation.check_validation_status(offer)
+    if booking_limit_datetime is None:
+        booking_limit_datetime = beginning
+
+    stock = Stock(
+        offer=offer,
+        beginningDatetime=beginning,
+        bookingLimitDatetime=booking_limit_datetime,
+        price=total_price,
+        numberOfTickets=number_of_tickets,
+        educationalPriceDetail=educational_price_detail,
+        quantity=1,
+    )
+    repository.save(stock)
+    logger.info("Educational stock has been created", extra={"offer": offer_id})
```

Comment:
```suggestion
    logger.info("Educational shadow stock has been created", extra={"offer": offer_id})
```
me paraît un peu mieux même :)